### DISCUSSION TITLE
create record with transaction doesn't return the properties directly.

### DISCUSSION THREAD

**Discussion by ricklang** [sequelize/sequelize#14218] [Help & Questions]
Sequelize 6.17.0, tedious 14.3.0.

When creating a new record without transaction:
```
let newUser = await db.User.create(input);
let userId = newUser.userId;    // This works
```

But if I add transaction, 

```
await db.sequelize.transaction(async (tx) => {
    let newUser = await db.User.create(input, { transaction: tx });
    let userId = newUser.userId;  // undefined
    let userId = newUser.dataValues.userId;   // This works
});
```

Why I can't read the column property directly when using transaction?

**Comment by ricklang** [comment]
Looks like if I move the variable declaration outside of transaction block, it is working as expected. Why?
```
let newUser;
await db.sequelize.transaction(async (tx) => {
    newUser = await db.User.create(input, { transaction: tx });
    let userId = newUser.userId;  // This works
});
```

**Reply by ephys** [reply]
This tells me the issue might not be with Sequelize but something unrelated that causes this weird behavior. I've personally never experienced this behavior in Sequelize before

**Comment by fzn0x** [comment]
I was suspicious it was related to _clsRun stuff it basically creates new context inside continuous local storage which will respond to only return dataValues instead of the return value of the create method.

**Reply by fzn0x** [reply]
Related: 
- ns.run https://github.com/sequelize/sequelize/blob/a21fc0cde41584521962a648c3820dc5c6551735/src/sequelize.js#L1070
- source: https://github.com/sequelize/sequelize/blob/a21fc0cde41584521962a648c3820dc5c6551735/src/sequelize.js#L1008
